manim/scene/scene.py

```diff
@@ -903,6 +771,81 @@ def wait_until(self, stop_condition, max_time=60):
         """
         self.wait(max_time, stop_condition=stop_condition)
 
+    def play_internal(self, *args, **kwargs):
+        """
+        This method is used to prep the animations for rendering,
+        apply the arguments and parameters required to them,
+        render them, and write them to the video file.
+
+        Parameters
+        ----------
+        *args : Animation or mobject with mobject method and params
+        **kwargs : named parameters affecting what was passed in *args e.g
+            run_time, lag_ratio etc.
+        """
+        if len(args) == 0:
+            warnings.warn("Called Scene.play with no animations")
+            return
+
+        animations = self.compile_play_args_to_animation_list(*args, **kwargs)
+        if (
+            len(animations) == 1
+            and isinstance(animations[0], Wait)
+            and not self.should_update_mobjects()
+        ):
+            self.add_static_frames(animations[0].duration)
+            return
+
+        moving_mobjects = None
+        static_mobjects = None
+        duration = None
+        stop_condition = None
+        time_progression = None
+        if not (len(animations) == 1 and isinstance(animations[0], Wait)):
+            # Paint all non-moving objects onto the screen, so they don't
+            # have to be rendered every frame
+            moving_mobjects, static_mobjects = self.get_moving_and_static_mobjects(
+                animations
+            )
+            self.renderer.save_static_mobject_data(self, static_mobjects)
+            time_progression = self.get_animation_time_progression(animations)
+        if len(animations) == 1 and isinstance(animations[0], Wait):
+            # TODO, be smart about setting a static image
+            # the same way Scene.play does
```

Comment:
Aren't thoses comments deprecated now? A static frame is now added at l. 796, when possible

Reply:
This comment is referring to instances where wait is called while some but not all Mobjects are moved by updaters. Line 796 only handles the case where wait is called and all Mobjects are stationary.